### PR TITLE
P4-1352 - Engage - New API endpoint to query Postmaster channels

### PR DESCRIPTION
**Pull request by rsdunya**
1. Standup an Engage instance, login as an Admin account.
2. Retrieve your API token from the /org/home/ page. api token
3. Execute a GET request to the ext/api/v2/channels.json endpoint 
`curl -H "Authorization: Token ********token*******" -X GET https://localhost/engage/ext/api/v2/channels.json`
Result: You should retrieve a list of all channels across all orgs.
ext api curl 

4. Using a non admin account, retry steps above.
Result: 
ext api curl 401

> {"detail":"You do not have permission to perform this action."}

### CODE REVIEW

File: temba/ext/api/v2/views.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            "title": "List Channels across all Orgs",
+            "url": reverse("ext.api.v2.channels"),
+            "slug": "channel-list",
+            "params": [
+                {
+                    "name": "uuid",
+                    "required": False,
+                    "help": "A channel UUID to filter by. ex: 09d23a05-47fe-11e4-bfe9-b8f6b119e9ab",
+                },
+                {"name": "address", "required": False, "help": "A channel address to filter by. ex: +250783530001"},
```
[the remark this answers is not in the document]
Comment (rsdunya) [reply temba/ext/api/v2/views.py]: @andrewkcarter updated, thanks!

**Comment by rsdunya** [comment]
> The org that the channel belongs to is a hard requirement. For every channel returned, Jeremy needs to know the org ID it belongs to.
> Also, an additional query parameter for Postmaster Device ID is required. The primary use case is so that Jeremy can check if a Postmaster device already has channels in Engage, and discover what org they exist in.

Is there a way we can modify the output from the queryset to concat the channel.org into the config object?

File: temba/ext/api/v2/views.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        # filter by address (optional)
+        address = params.get("address")
+        if address:
+            queryset = queryset.filter(address=address)
+
+        c_type = params.get("type")
+        if c_type:
+            queryset = queryset.filter(channel_type=c_type)
+
+        c_address = params.get("device")
```
Comment (andrewkcarter) [inline temba/ext/api/v2/views.py]: We talked about using `device_id` consistently through Engage, Post Office, and Postmaster. Please update this parameter name. In a separate PR, can you update the Postmaster Channel config and UI form to also refer to the value as `device_id`/`Device ID` for consistency?

File: temba/ext/api/v2/views.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            "url": reverse("ext.api.v2.channels"),
+            "slug": "channel-list",
+            "params": [
+                {
+                    "name": "uuid",
+                    "required": False,
+                    "help": "A channel UUID to filter by. ex: 09d23a05-47fe-11e4-bfe9-b8f6b119e9ab",
+                },
+                {"name": "address", "required": False, "help": "A channel address to filter by. ex: +250783530001"},
+                {"device": "device id", "required": False, "help": "A postmaster Device ID  to filter by. ex: pm_whatsapp_1"},
```
Comment (andrewkcarter) [inline temba/ext/api/v2/views.py]: `devide_id`

File: temba/ext/api/v2/views.py (hunk 2 above)
Comment (rsdunya) [reply temba/ext/api/v2/views.py]: Yessir

File: temba/ext/api/v2/views.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        # filter by address (optional)
+        address = params.get("address")
+        if address:
+            queryset = queryset.filter(address=address)
+
+        c_type = params.get("type")
+        if c_type:
+            queryset = queryset.filter(channel_type=c_type)
+
+        c_address = params.get("device")
```
Comment (andrewkcarter) [inline temba/ext/api/v2/views.py]:
```suggestion
        c_address = params.get("device_id")
```

File: temba/ext/api/v2/views.py [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+            "url": reverse("ext.api.v2.channels"),
+            "slug": "channel-list",
+            "params": [
+                {
+                    "name": "uuid",
+                    "required": False,
+                    "help": "A channel UUID to filter by. ex: 09d23a05-47fe-11e4-bfe9-b8f6b119e9ab",
+                },
+                {"name": "address", "required": False, "help": "A channel address to filter by. ex: +250783530001"},
+                {"device": "device id", "required": False, "help": "A postmaster Device ID  to filter by. ex: pm_whatsapp_1"},
```
Comment (andrewkcarter) [inline temba/ext/api/v2/views.py]:
```suggestion
                {"device_id": "device id", "required": False, "help": "A postmaster Device ID  to filter by. ex: pm_whatsapp_1"},
```

**Pull request merged**